runtime/strategies/coalesce-recipes.js

```diff
@@ -32,6 +31,47 @@ export class CoalesceRecipes extends Strategy {
     await index.ready;
 
     return Recipe.over(this.getResults(inputParams), new class extends Walker {
+      onSlot(recipe, slot) {
+        // Find slots that according to their provided-spec must be consumed, but have no consume connection.
+        if (slot.consumeConnections.length > 0) {
+          return; // slot has consume connections.
+        }
+        if (!slot.sourceConnection || !slot.sourceConnection.slotSpec.getProvidedSlotSpec(slot.name).isRequired) {
+          return; // either a remote slot (no source connection), or a not required one.
+        }
+
+        let results = [];
+
+        index.findConsumeSlotConnectionMatch(slot).forEach(({slotConn, matchingHandles}) => {
+          results.push((recipe, slot) => {
+            let {cloneMap} = slotConn.recipe.mergeInto(slot.recipe);
+            let mergedSlotConn = cloneMap.get(slotConn);
+            mergedSlotConn.connectToSlot(slot);
+            matchingHandles.forEach(({handle, matchingConn}) => {
+              // matchingConn in the mergedSlotConnection's recipe should be connected to `handle` in the slot's recipe.
+              let mergedMatchingConn = cloneMap.get(matchingHandles[0].matchingConn);
```

Comment:
matchingHandles[0].matchingConn looks wrong in the forEach, shouldn't it just be matchingConn?

Reply:
good catch, thanks! done